request-cert/main.go

```diff
@@ -52,53 +51,103 @@ func main() {
 
 	// Check certificate type.
 	var template *x509.CertificateRequest
-	var filePrefix string
+	var filename, csrName string
 	var wantServerAuth bool
 
+	hostname, err := os.Hostname()
```

Comment:
Changed csr names (but not filenames or secret names) to include the namespace. eg: `default:client.root`.

Here's a dump of most of the built (with k8s stuff removed, only showing cockroach/example):
```
$ kubectl create -f cockroachdb-statefulset-secure.yaml
$ kubectl create -f example_app_secure.yaml
$ kubectl create -f example_app_secure.yaml --namespace kv-client

$ kubectl get all --all-namespaces  && printf "\nCSRs:\n" && kubectl get csr && printf "\nSecrets:\n" && kubectl get secret --all-namespaces
NAMESPACE     NAME                                 READY     STATUS             RESTARTS   AGE
default       po/cockroachdb-0                     1/1       Running            0          15m
default       po/cockroachdb-1                     1/1       Running            0          8m
default       po/cockroachdb-2                     1/1       Running            0          8m
default       po/example-secure-3203699586-jgmwb   1/1       Running            0          4m
kv-client     po/example-secure-3203699586-2dhm4   0/1       CrashLoopBackOff   4          3m

NAMESPACE     NAME                       CLUSTER-IP   EXTERNAL-IP   PORT(S)              AGE
default       svc/cockroachdb            None         <none>        26257/TCP,8080/TCP   16m
default       svc/cockroachdb-public     10.0.0.188   <none>        26257/TCP,8080/TCP   16m

NAMESPACE   NAME                       DESIRED   CURRENT   AGE
default     statefulsets/cockroachdb   3         3         16m

NAMESPACE     NAME                    DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
default       deploy/example-secure   1         1         1            1           4m
kv-client     deploy/example-secure   1         1         1            0           3m

NAMESPACE     NAME                           DESIRED   CURRENT   READY     AGE
default       rs/example-secure-3203699586   1         1         1         4m
kv-client     rs/example-secure-3203699586   1         1         0         3m

CSRs:
NAME                         AGE       REQUESTOR                                 CONDITION
default.client.root          4m        system:serviceaccount:default:default     Approved,Issued
default.node.cockroachdb-0   15m       system:serviceaccount:default:default     Approved,Issued
default.node.cockroachdb-1   8m        system:serviceaccount:default:default     Approved,Issued
default.node.cockroachdb-2   8m        system:serviceaccount:default:default     Approved,Issued
kv-client.client.root        3m        system:serviceaccount:kv-client:default   Approved,Issued

Secrets:
NAMESPACE     NAME                         TYPE                                  DATA      AGE
default       default.client.root          Opaque                                2         4m
default       default.node.cockroachdb-0   Opaque                                2         8m
default       default.node.cockroachdb-1   Opaque                                2         8m
default       default.node.cockroachdb-2   Opaque                                2         5m
kv-client     kv-client.client.root        Opaque                                2         1m
```

CSRs are separated due to namespace being included in the name. Secrets are already by namespace, but I like having the identical name.
The crashloop is on failing to lookup `cockroachdb-public`, not CSRs or secrets.